src/server/api/AdminPanel.ts

```diff
@@ -26,7 +24,26 @@ const validCommands = [
 
 function processCommand(req: CustomRequest, res: Response): void {
 	const command = req.params["command"]!;
-	const commandAndArgs = command.split(" ");
+	const commandAndArgs: string[] = [];
+	let inQuote: boolean = false;
+	let temp: string = "";
+	for (let i = 0; i < command.length; i++) {
```

Comment:
This logic is great! Could you just move it to an external function, named `getArgumentsFromCommand()` or something?